### ISSUE TITLE
[Feature Request] Find all definitions for a key in Meta

### ISSUE THREAD

**Issue by amarjeet000** [BetterThanTomorrow/calva#488]
It would be great if there is a facility to find all the definitions with the help of a key in `meta`. For example, finding all definitions where `meta` has `:todo` key.

```
(defn
  ^{:doc ""
     :todo ""}
  my-fn
  [params]
  (body))
```
It is possible to improvise on this thought, such as providing enhanced support for some pre-defined meta keys. For example, if `:api` is a natively-supported key, then it is possible to provide feature to search for all definitions with `api` as `1.2.0`.

**Comment by bpringe** [comment]
This is interesting

**Comment by bpringe** [comment]
Looking at this again, I think vs code's search features provide enough functionality to find things like this. Please comment if you disagree.

**Comment by bpringe** [comment]
To expand on the above, it's probably possible to do, but the effort involved to make something only slightly better than vs code's search (imo) would probably never be worth it versus other feature additions. However, if it's something you'd use yourself a lot, PRs are definitely welcome.

**Comment by amarjeet000** [comment]
You are right that the trade-off w.r.t time investment on this feature compared to on other important ones might not make sense. I think the core purpose of this proposal can be addressed by the search feature of VS Code itself. I would put my other interests related to this feature as secondary.

**Comment by PEZ** [comment]
We can think of this a bit more generally, maybe. What if we had a sort of query language that could find things for us in a code base in a structural way? A bit like `get-in`... Then this meta-key search could be a query that can be shared or even come default with Calva. Maybe it is not feasible, but it might be fun exploring this path a bit. 😄

**Comment by amarjeet000** [comment]
I am going to upvote your idea Peter, even though its not gonna be taken up 😄

**Comment by bpringe** [comment]
Interesting idea
